src/extension/CMakeLists.txt

```diff
@@ -14,45 +14,192 @@ add_library(thrive_extension SHARED
   core/GodotJoltConversions.hpp
 )
 
-if(MSVC)
-  target_compile_options(thrive_extension PRIVATE /W4 /wd4068)
+target_sources(thrive_extension PRIVATE ${EXTENSION_SOURCES})
 
-  if(WARNINGS_AS_ERRORS)
-    target_compile_options(thrive_extension PRIVATE /WX)
-  endif()
+# Print files and configure build
+print_source_files(thrive_extension)
 
-  # TODO debug symbols in release for MSVC (and Distribution mode)
+# Common compile definitions for both OS
+target_compile_definitions(thrive_extension PRIVATE 
+    THRIVE_NATIVE_BUILD
+    _HAS_EXCEPTIONS=1
+    JPH_PROFILE_ENABLED=0
+    JPH_DEBUG_RENDERER=0
```

Comment:
Why is this here? When compiling in debug mode this has to be defined to one or everything will break (well at least for the thrive native library, but it would be super weird to not have consistent options).